### PR TITLE
Zeppelin: Compress the image by doing the build on a separate image

### LINKED ISSUE #17231: Zeppelin image is huge
The `gcr.io/google_containers/zeppelin` image added by #17025 is over a gig. Investigate how to trim this down. (Cut out interpretters? Make them more modular somehow, possibly composing them once we have some solution like #831?) It's not even clear there's upstream support for making this more modular, though.

### PR DESCRIPTION
**Pull request by zmerlynn**
Pull the tar through to the image we need it on.

Starts to fix #17231, but to fix that fully will probably require upstream changes.

### CODE REVIEW

**Comment by zmerlynn** [comment]
cc @timothysc @timstclair

File: examples/spark/README.md [hunk 1]
```diff
@@ -290,9 +290,6 @@ Then visit [http://localhost:8080/](http://localhost:8080/).
 
 ## Known Issues With Zeppelin
 
-* The Zeppelin pod is large, so it may take a while to pull depending on your
```
Comment (timothysc) [inline examples/spark/README.md]: Umm, it's still pretty large.  What's the before after size now?

File: examples/spark/images/zeppelin-build/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,50 @@
+# Copyright 2015 The Kubernetes Authors All rights reserved.
```
Comment (timothysc) [inline examples/spark/images/zeppelin-build/Dockerfile]: Probably want to denote somewhere that the whole purpose of this image is simply to produce the other image.

**Comment by timothysc** [comment]
So I see what you're doing here, and I've seen it done elsewhere. 
I just debate if it's zepplin specific, or if you'll grow your build image to potentially produce multiple pieces of a stack...

File: examples/spark/images/zeppelin-build/Dockerfile (hunk 2 above)
Comment (zmerlynn) [inline examples/spark/images/zeppelin-build/Dockerfile]: Done.

**Comment by zmerlynn** [comment]
I went ahead and reverted the `README.md` change as well.

As for making it more generic, let's use the "one, two, many" principal and refactor when the pattern appears, if I find myself doing it yet again. I certainly think it's possible.

**Comment by zmerlynn** [comment]
@timothysc: PTAL. I think to fully fix this, we'll need upstream changes, since Zeppelin itself is huge.

File: examples/spark/zeppelin-controller.yaml [hunk 3]
```diff
@@ -13,7 +13,7 @@ spec:
     spec:
       containers:
         - name: zeppelin
-          image: gcr.io/google_containers/zeppelin:v0.5.5_v1
+          image: gcr.io/google_containers/zeppelin:v0.5.5_v2
```
Comment (timothysc) [inline examples/spark/zeppelin-controller.yaml]: Is this tag the standard way we're going to version images?  Dunno if there is convention here.

**Comment by zmerlynn** [comment]
@k8s-bot unit test this

File: examples/spark/zeppelin-controller.yaml (hunk 3 above)
Comment (zmerlynn) [inline examples/spark/zeppelin-controller.yaml]: It's similar to how the Spark containers were tagged originally so I followed it. Is there a better scheme for "given semver vX.Y.Z[-foo], this is docker image N?" We should probably standardize on something.

Comment (timothysc) [inline examples/spark/zeppelin-controller.yaml]: Agreed, we may want to start curating a curation "punchlist" for the future.

**Comment by timothysc** [comment]
lgtm on passing tests.  There are some broader questions to start asking on formalized curation as we branch out.  

- Tagging 
- Build image handling 
...

**Comment by zmerlynn** [comment]
@k8s-bot test this

**Comment by zmerlynn** [comment]
@k8s-oncall: @k8s-bot seems grumpy

**Comment by zmerlynn** [comment]
@k8s-bot test this

File: examples/spark/zeppelin-controller.yaml (hunk 3 above)
Comment (zmerlynn) [inline examples/spark/zeppelin-controller.yaml]: BTW, "curating a curation" amused me.

**Comment by zmerlynn** [comment]
Hand merging. This only effects the image, and I already pushed it.

**Pull request merged**